packages/codemods/src/helpers/result.ts

```diff
@@ -0,0 +1,127 @@
+import { Chainable, Flattened } from './chainable';
+
+export interface Ok<R, E> extends ResultInternal<R, E> {
+  ok: true;
+  value: R;
+}
+
+export interface Err<R, E> extends ResultInternal<R, E> {
+  ok: false;
+  value: E;
+}
+
+class ResultInternal<R, E> implements Chainable<R> {
+  public ok: boolean;
+  public value: R | E;
+
+  public constructor(options: { ok: boolean; value: R | E }) {
+    this.ok = options.ok;
+    this.value = options.value;
+  }
+
+  public map<T>(this: Result<R, E>, fn: (v: R) => T): Result<T, E> {
```

Comment:
why do we need `chain`? isn't `map` alone enough?

Reply:
Map is sufficient, but I think that chain is a nice to have. Take the instance where I have a function like `processString(str): Result<string, E>`  and I have some result like `fetchedString: Result<string, E>` that was potentially obtained from a rest call. I want to process the string and I cannot do `fetchedString.map(processString)` without calling flatten afterwards. Having something like chain to just handle that is nice. There are also cases where you might want it to be nested, in which case having map is still useful.

Reply:
just pointing out couple things for you to consider:
1. the names `map` and `chain` really confuse me.  I think in the world we only have one of them, either name works for me. having both exist forced me to read into implementation to understand the difference
2. you have `orChain`, not `orMap`